### PR TITLE
perf: add kzg.UnsafeToBytes and kzg.UnsafeFromBytes methods

### PR DESCRIPTION
**Pull request by gbotrel**
# Description

TLDR; adds a method (no breaking changes) to `kzg.SRS` object that quickly dumps a binary representation without doing any checks, and optionally, enables to truncate the SRS at writing / reading time. `UnsafeFromBytes` is 6x time faster than `UnsafeReadFrom` 

godoc:
```
// UnsafeToBytes returns the binary encoding of the entire SRS memory representation
// It is meant to be use to achieve fast serialization/deserialization and
// is not compatible with WriteTo / ReadFrom. It does not do any validation
// and doesn't encode points in a canonical form.
// @unstable: the format may change in the future
// If maxPkPoints is provided, the number of points in the ProvingKey will be limited to maxPkPoints
func (srs *SRS) UnsafeToBytes(maxPkPoints ...int) ([]byte, error) {
```

benchmark, 1 <<24 points bls12377
```
BenchmarkSerializeSRS/WriteTo-10                       1        1183526917 ns/op        2147545088 B/op      550 allocs/op
BenchmarkSerializeSRS/WriteRawTo-10                    1        1249923583 ns/op        4295028640 B/op      542 allocs/op
BenchmarkSerializeSRS/UnsafeToBytes-10                 4         287003656 ns/op        1610821252 B/op      522 allocs/op
BenchmarkDeserializeSRS/UnsafeReadFrom-10                      1        2677718292 ns/op        5922632568 B/op     1597 allocs/op
BenchmarkDeserializeSRS/UnsafeFromBytes-10                     3         419717778 ns/op        2147761242 B/op     1205 allocs/op
```

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I did not modify files generated from templates
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: internal/generator/kzg/template/marshal.go.tmpl [hunk 1]
```diff
@@ -72,6 +75,88 @@ func (vk *VerifyingKey) writeTo(w io.Writer, options ...func(*{{.CurvePackage}}.
...
+	if len(maxPkPoints) == 1 && maxPkPoints[0] > 0 && int(nbPoints) > maxPkPoints[0] {
+		nbPoints = uint64(maxPkPoints[0])
+	}
+
+	srs.Pk.G1 = make([]{{.CurvePackage}}.G1Affine, nbPoints)
+
+	// read the limbs directly
+	var bbuf [fp.Bytes * 2]byte
+	for i := 0; i < int(nbPoints); i++ {
+		if _, err := io.ReadFull(buf, bbuf[:]); err != nil {
```
Comment (ivokub) [inline internal/generator/kzg/template/marshal.go.tmpl:147]: `srs.Vk.ReadFrom` returns number of bytes read and it is 8 byte for the length. Instead of `io.ReadFull` we can just set `bbuf = buf[n+8+i*2*fp.Bytes:n+8+(i+1)*2*fp.Bytes]`

File: internal/generator/kzg/template/kzg.test.go.tmpl [hunk 2]
```diff
@@ -606,6 +639,109 @@ func BenchmarkToLagrangeG1(b *testing.B) {
...
+	// 		var newSRS SRS
+	// 		_, err := newSRS.ReadFrom(bytes.NewReader(buf.Bytes()))
+	// 		if err != nil {
+	// 			b.Fatal(err)
+	// 		}
+	// 	}
+	// })
+
+	b.Run("UnsafeReadFrom", func(b *testing.B) {
+		b.ResetTimer()
```
Comment (ivokub) [inline internal/generator/kzg/template/kzg.test.go.tmpl:715]: Should reset timer after writing two lines below? Otherwise skews the result a bit.

File: internal/generator/kzg/template/kzg.test.go.tmpl [hunk 3]
```diff
@@ -606,6 +639,109 @@ func BenchmarkToLagrangeG1(b *testing.B) {
...
+			var newSRS SRS
+			_, err := newSRS.UnsafeReadFrom(bytes.NewReader(buf.Bytes()))
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
+
+	b.Run("UnsafeFromBytes", func(b *testing.B) {
+		b.ResetTimer()
```
Comment (ivokub) [inline internal/generator/kzg/template/kzg.test.go.tmpl:731]: Reset timer after `srs.UnsafeToBytes`

**Review by ivokub — APPROVED** [review]
Small comments, but otherwise looks great.
